src/components/scenes/SwapSuccessScene.tsx

```diff
@@ -1,97 +1,82 @@
-import { Disklet } from 'disklet'
 import * as React from 'react'
 import { View } from 'react-native'
 import ConfettiCannon from 'react-native-confetti-cannon'
 
+import { useHandler } from '../../hooks/useHandler'
 import { lstrings } from '../../locales/strings'
-import { connect } from '../../types/reactRedux'
+import { useSelector } from '../../types/reactRedux'
 import { EdgeSceneProps } from '../../types/routerTypes'
 import { needToShowConfetti } from '../../util/show-confetti'
 import { SceneWrapper } from '../common/SceneWrapper'
 import { showError } from '../services/AirshipInstance'
-import { cacheStyles, Theme, ThemeProps, withTheme } from '../services/ThemeContext'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
 import { EdgeText } from '../themed/EdgeText'
 import { Fade } from '../themed/Fade'
 import { MainButton } from '../themed/MainButton'
 
-interface OwnProps extends EdgeSceneProps<'swapSuccess'> {}
-
-interface StateProps {
-  userId: string
-  disklet: Disklet
-}
-
-interface LocalState {
-  showButton: boolean
-  showConfetti: boolean
-}
-
-type Props = StateProps & OwnProps & ThemeProps
+interface Props extends EdgeSceneProps<'swapSuccess'> {}
 
 const confettiProps = {
   count: 250,
   origin: { x: -50, y: -50 },
   fallSpeed: 4000
 }
 
-export class SwapSuccessSceneComponent extends React.PureComponent<Props, LocalState> {
-  constructor() {
-    // @ts-expect-error
-    super()
-    this.state = { showButton: false, showConfetti: false }
-  }
+export const SwapSuccessScene = (props: Props) => {
+  const { navigation } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
 
-  componentDidMount(): void {
-    this.showConfetti().catch(err => showError(err))
-  }
+  const [showButton, setShowButton] = React.useState(false)
+  const [showConfetti, setShowConfetti] = React.useState(false)
 
-  done = () => {
-    const { navigation } = this.props
-    this.setState({ showButton: false })
+  const userId = useSelector(state => state.core.account.id)
+  const disklet = useSelector(state => state.core.disklet)
+
+  const done = useHandler(() => {
+    setShowButton(false)
     navigation.navigate('swapTab', { screen: 'swapCreate' })
-  }
+  })
 
-  showConfetti = async () => {
-    const { userId, disklet } = this.props
+  const showConfettiAsync = useHandler(async () => {
     const show: boolean = await needToShowConfetti(userId, disklet)
 
     if (show) {
-      this.setState({ showConfetti: true })
+      setShowConfetti(true)
       setTimeout(() => {
-        this.setState({ showButton: true })
+        setShowButton(true)
       }, 4500)
     } else {
-      this.setState({ showButton: true })
+      setShowButton(true)
     }
-  }
+  })
 
-  renderConfetti() {
-    if (!this.state.showConfetti) return null
+  React.useEffect(() => {
+    showConfettiAsync().catch(err => showError(err))
```

Comment:
Just put the code for showConfettiAsync inside the useEffect or useAsyncAffect. That avoids declaring the function on every render and the overhead of useHandler.